substrate/client/network/src/request_responses.rs

```diff
@@ -594,6 +618,60 @@ impl NetworkBehaviour for RequestResponsesBehaviour {
 
 	fn poll(&mut self, cx: &mut Context) -> Poll<ToSwarm<Self::ToSwarm, THandlerInEvent<Self>>> {
 		'poll_all: loop {
+			// Poll the periodic request check.
+			if let Poll::Ready(_) = self.periodic_request_check.poll_tick(cx) {
+				// First pass is needed to collect timed out requests and
+				// submit `OutboundFailure::Timeout` on the moved oneshot sender.
+				let timedout_requests: Vec<_> = self
+					.pending_requests
+					.iter()
+					.filter_map(|(id, req)| {
+						let Some(ProtocolDetails { request_timeout, .. }) =
+							self.protocols.get(&id.protocol)
+						else {
+							log::warn!(
+								target: "sub-libp2p",
+								"Request with id {:?} has no protocol registered.",
+								id.request_id,
+							);
+							return Some(id.clone())
+						};
+
+						let elapsed = req.started_at.elapsed();
+						if elapsed > *request_timeout {
+							Some(id.clone())
+						} else {
+							None
+						}
+					})
+					.collect();
+
+				if !timedout_requests.is_empty() {
+					log::warn!(
+						target: "sub-libp2p",
+						"Requests {timedout_requests:?} force detected as timeout.",
```

Comment:
can we print the requests and the protocol ?

Reply:
Yep, this is the ProtocolRequestID which contains both the libp2p requestID ++ protocol name

https://github.com/paritytech/polkadot-sdk/blob/c3b01dac9a1f582b7e5724534fb4325e72bcb11e/substrate/client/network/src/request_responses.rs#L331-L334